examples/virtual_terminal/aux_inputs/main.cpp

```diff
@@ -0,0 +1,199 @@
+#include "isobus/hardware_integration/available_can_drivers.hpp"
+#include "isobus/hardware_integration/can_hardware_interface.hpp"
+#include "isobus/isobus/can_general_parameter_group_numbers.hpp"
+#include "isobus/isobus/can_network_manager.hpp"
+#include "isobus/isobus/can_partnered_control_function.hpp"
+#include "isobus/isobus/isobus_virtual_terminal_client.hpp"
+#include "isobus/utility/iop_file_interface.hpp"
+#include "isobus/utility/system_timing.hpp"
+
+#include "console_logger.cpp"
+#include "object_pool_ids.h"
+
+#include <atomic>
+#include <csignal>
+#include <iostream>
+#include <memory>
+
+//! It is discouraged to use global variables, but it is done here for simplicity.
+static std::shared_ptr<isobus::VirtualTerminalClient> TestVirtualTerminalClient = nullptr;
+static constexpr std::uint64_t MODEL_IDENTIFICATION_CODE = 1; ///< The model identification code of 'our' input device, this should be increased when changes are made to the input(s) definitions
+
+static constexpr std::uint64_t BUTTON_CYCLIC_DELAY = 3500; ///< 1 second between button presses
+static constexpr std::uint64_t B2_CYCLIC_DELAY = 1000; ///< 3.5 seconds between slider movements
+static std::uint64_t lastButtonTimestamp = 0;
+static std::uint64_t lastSliderTimestamp = 0;
+
+static bool buttonPressed = false;
+static std::uint16_t buttonTransitions = 0;
+
+static constexpr std::uint16_t SLIDER_MAX_POSITION = 0xFAFF;
+static bool backToZero = false;
+static std::uint16_t sliderPosition = 0;
+static std::atomic_bool running = { true };
+
+using namespace std;
+
+void signal_handler(int)
+{
+	running = false;
+}
+
+void simulate_button_press()
+{
+	buttonPressed = !buttonPressed;
+	TestVirtualTerminalClient->update_auxiliary_input(AUXN_INPUT_BUTTON, buttonPressed, buttonTransitions);
+	// std::cout << "Auxiliary Button input value change send (pressed: " << buttonPressed << ", transitions: " << buttonTransitions << ")" << std::endl;
```

Comment:
If this commented out code is just for debugging the example we may want to remove it, same goes for some other spots in this file.
Re: Autosar Rule A2-8-2 Sections of code shall not be “commented out”.